src/cogs/handler.js

```diff
@@ -0,0 +1,138 @@
+        Key: 'Adobe.ArchPath',
+        Values: ['EC.SpaceCat.Services'],
+        MatchOptions: ['EQUALS'],
+      },
+    },
+    Metrics: [
+      'UnblendedCost',
+    ],
+    GroupBy: [
+      {
+        Key: 'SERVICE',
+        Type: 'DIMENSION',
+      },
+      {
+        Key: 'Environment',
+        Type: 'TAG',
+      },
+    ],
+  };
+}
+
+/**
+ * This function parses the date in YYYY/MM/DD format like 2021/01/01 and return the date in
+ * @param {string} date format like 2021-01-01
+ * @returns string like Jan-21
+ */
+const parseYearDate = (date) => {
+  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+  const yrMonth = date.split(/-/);
+  return `${months[yrMonth[1] - 1]}-${yrMonth[0].substring(2)}`;
+};
+
+/**
+ * This function maps the AWS service name to the service name used in SpaceCat
+ * @param {string} service AWS service name
+ * @returns string like LAMBDA
+ */
+const serviceMap = (service) => {
+  switch (service) {
+    case 'AWS Lambda':
+      return 'LAMBDA';
+    case 'AWS Secrets Manager':
+      return 'SECRETS MANAGER';
+    case 'Amazon DynamoDB':
+      return 'DYNAMODB';
+    case 'Amazon Simple Storage Service':
+      return 'S3';
+    case 'Amazon Simple Queue Service':
+      return 'SQS';
+    case 'AmazonCloudWatch':
+      return 'CLOUDWATCH';
+    default:
+      return service;
+  }
+};
+
+/**
+ * This function processes the response from AWS Cost Explorer API
+ * @param {Object} data response from AWS Cost Explorer API
+ * @returns {Object} results by month and Year wise.
+ */
+function processAWSResponse(data) {
+  if (data && data.ResultsByTime && data.ResultsByTime.length > 0) {
+    const result = {};
+    data.ResultsByTime.forEach((r) => {
+      if (r.Groups && r.Groups.length > 0) {
+        const granularity = {};
+        r.Groups.forEach((group) => {
+          const key = group.Keys[0];
+          if (group.Metrics && group.Metrics.UnblendedCost) {
+            // eslint-disable-next-line max-len
+            granularity[serviceMap(key)] = parseFloat(group.Metrics.UnblendedCost.Amount).toFixed(2);
+          }
+        });
+        result[parseYearDate(r.TimePeriod.Start)] = granularity;
+      }
+    });
+    return result;
+  }
+  return {};
+}
+export default async function auditCOGs(message, context) {
+  const { type, startDate, endDate } = message;
+  const { log, sqs } = context;
+  const { region } = context.runtime;
+  log.info(`Fetching Cost Usage from ${startDate} to ${endDate}`);
+
+  try {
+    const client = new CostExplorerClient({ region });
+    const input = buildAWSInput(startDate, endDate);
+    const command = new GetCostAndUsageCommand(input);
+    const response = await client.send(command);
+    const usageCost = processAWSResponse(response);
+    if (Object.keys(usageCost).length === 0) {
+      return notFound('No Cost Usage found');
+    }
+    Object.keys(usageCost).forEach(async (monthYear) => {
```

Comment:
@sudo-buddy, how does this work when the `startDate` & `endDate` variables span through multiple months partially? For example, `startDate=2024-01-15` and `endDate=2024-02-06`?

Reply:
@ekremney, the requirement is to get the result by month wise so my assumption was alway get the result for previous whole month.